distributed/shuffle/tests/test_shuffle.py

```diff
@@ -524,104 +534,6 @@ async def test_crashed_worker_during_unpack(c, s, a):
         await clean_scheduler(s)
 
 
-class BlockedRegisterCompleteShuffleWorkerExtension(ShuffleWorkerExtension):
-    def __init__(self, worker: Worker) -> None:
-        super().__init__(worker)
-        self.in_register_complete = asyncio.Event()
-        self.block_register_complete = asyncio.Event()
-
-    async def _register_complete(self, shuffle: Shuffle) -> None:
-        self.in_register_complete.set()
-        await super()._register_complete(shuffle)
-        await self.block_register_complete.wait()
-
-
-@pytest.mark.parametrize("kill_barrier", [True, False])
-@gen_cluster(
-    client=True,
-    worker_kwargs={
-        "extensions": {"shuffle": BlockedRegisterCompleteShuffleWorkerExtension}
-    },
-    nthreads=[("", 1)] * 2,
-)
-async def test_closed_worker_during_final_register_complete(c, s, a, b, kill_barrier):
-
-    df = dask.datasets.timeseries(
```

Comment:
I assume these specific race conditions are no longer possible or how come this is being deleted?

Reply:
We removed `register_complete` and changed the logic around cleanup, so this race is impossible.